README.md

```diff
@@ -0,0 +1,46 @@
+# Honeybadger for Crystal
+
+HTTP::Handler and exception notifier for the [⚡Honeybadger error notifier](https://www.honeybadger.io/).
+
+## Getting Started
+
+Update your `shard.yml` to include honeybadger:
+
+```diff
+dependencies:
++  honeybadger:
++    github: honeybadger-io/honeybadger-crystal
+```
+
+Add the Honeybadger::Handler to the HTTP::Server stack:
+
+```crystal
+honeybadger_api_key = ENV["HONEYBADGER_API_KEY"]? || "00000000"
+honeybadger_enabled = MyServer.production?
+
+Honeybadger.configure(api_key: honeybadger_api_key)
+
+Honeybadger::Handler.new(
+  enabled: honeybadger_enabled,
```

Comment:
I think I'd prefer if `enabled` was a global config option, and let's actually call it `report_data` (default `true`):

```crystal
Honeybadger.configure(
  api_key: 'API key',
  report_data: true
)
```

In the future we'll also want to add global `environment` (String, default: `Crystal.env` if there is such a thing, otherwise `nil`) and `development_environments` (array, default: `["test", "development"]`) config options, like our Ruby gem has. The default value for `report_data` would then become dynamic: `development_environments.include?(environment)` (but could be explicitly overridden by the user).

Reply:
**Some Background**
For better or for worse, crystal community [has discussed and decided](https://github.com/crystal-lang/crystal/issues/5220) not to implement an official Crystal.env paradigm -- though there is a [dependency](https://github.com/crystal-lang/crystal-env) which is contributed by a language core-team member. It hasn't seen much adoption. Which means it's up to the (web) framework developers to implement something along these lines if needed. It doesn't preclude this shard from having the features you've outlined. The defacto standard is one which matches Rails - "development, test, production", but the source of truth differs according to the application framework.

The [dispatcher](https://github.com/honeybadger-io/honeybadger-crystal/blob/server_context_dispatch/src/honeybadger/dispatch.cr#L7) is what's responsible for actually sending the request to the API, and `enabled` is right now a per-instance variable, which gets handed down from the handler. Moving it up to the module level makes sense because it's unlikely to be switched on and off inside a running app.

Reply:
@robacarp makes sense, thanks for the background, that was a great read. If you want to just move and rename the `enabled` option for now, I can create an issue for adding the `development_environments` config which we can tackle after this PR is merged.